File: work-spring/day0903_jwt/src/main/java/com/grepp/jwt/controller/UserController.java
package com.grepp.jwt.controller;

import com.grepp.jwt.model.dto.UserDTO;
import com.grepp.jwt.model.entity.UserEntity;
import com.grepp.jwt.model.service.UserService;
import com.grepp.jwt.util.MyJwtTokenProvider;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/auth")
public class UserController {
    @Autowired
    private UserService userService;
    @Autowired
    private MyJwtTokenProvider myJwtTokenProvider; // 토큰 생성 및 유효성 검증

    @PostMapping("/signup")
    public ResponseEntity<?> register(@RequestBody UserDTO userDTO){
        // 클라이언트 쪽에서 유효한 값을 '전송' 제대로 했는지 체크
        if(userDTO == null || userDTO.getPassword()==null) throw new RuntimeException("Invalid Password !!!!");

        UserEntity entity = userDTO.toEntity();
        UserEntity resultEntity = userService.join(entity);
        return ResponseEntity.ok().body(new UserDTO(resultEntity));
    }

    @PostMapping("/signin")
    public ResponseEntity<?> signin(@RequestBody UserDTO userDTO){
        UserEntity userEntity = userService.getByCredentials(userDTO.getUsername(), userDTO.getPassword());

        if(userEntity != null){ // login success
            String token = myJwtTokenProvider.createMyToken(userEntity, 1000*60);
            userDTO.setToken(token);
            return ResponseEntity.ok().body(userDTO); // token이 포함된 정보가 응답됨.
        }
        return ResponseEntity.badRequest().body("Login Failed");
    }
}

File: work-kotlin/day1021_todo/src/main/kotlin/com/grepp/todo/controller/UserController.kt
package com.grepp.todo.controller

import com.grepp.todo.model.dto.UserDTO
import com.grepp.todo.model.entity.UserEntity
import com.grepp.todo.model.service.UserService
import com.grepp.todo.util.MyJwtTokenProvider
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/auth")
class UserController (val service:UserService){
    // autowired는 객체 생성 이후 주입이기 때문에 반드시 lateinit var 해줘야 함.
    @Autowired
    lateinit var myTokenProvider : MyJwtTokenProvider
    
    // 회원가입
    @PostMapping("/signup")
    fun register(@RequestBody userDTO : UserDTO): ResponseEntity<UserEntity> {
        val result = service.join(userDTO.toEntity())
        println("join result : ${result}")
        result.role="ADMIN" // 지금 권한이나
        result.authProvider="grepp" // 소셜 정보 하나도 구현 안해서 ;;; 임의로 ;;; json 만들어서 응답 못한다 해서 ;;
        return ResponseEntity.ok().body(result)
    }
    
    // 로그인
    @PostMapping("/login")
    fun login(@RequestBody userDTO: UserDTO):ResponseEntity<Any>{
        val userEntity = service.login(userDTO.username, userDTO.password)

        println("login result : ${userEntity}")
        if(userEntity!=null){ // 로그인 성공했어? 그럼 토큰 발급해줘야지~~
            val token = myTokenProvider.createMyToken(userEntity,1000*60*2)
            userDTO.token = token
            return ResponseEntity.ok().body(userDTO)
        }

        return ResponseEntity.badRequest().body("login failed!!")
    }
}